### PR TITLE
docs: add streaming cost

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

add streaming cost in doc

Screenshot 2024-08-01 at 11 13 17

Screenshot 2024-08-01 at 11 13 42

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1560)
###### _updated: 7/30/2024, 08:52:25 (UTC+0)_

**Comment by aws-solution-clickstream-analytics** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1560&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1560&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1560&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

File: docs/en/plan-deployment/cost.md [hunk 1]
```diff
@@ -119,6 +119,22 @@ Key assumptions include:
 !!! info "Note"
     The term **Redshift storage cost** refers to the cost of Redshift storage incurred for one month based on the corresponding RPS (Requests Per Second) specified in the above table. If the data is stored for more than one month, please refer to the [Redshift pricing](https://aws.amazon.com/redshift/pricing/?nc1=h_ls) for calculating the pricing.
 
+## Realtime streaming mode
+
+If you enable real-time reports, the following additional cost will be incurred:
+Key assumptions include:
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:125]: need new line?

File: docs/en/plan-deployment/cost.md [hunk 2]
```diff
@@ -119,6 +119,22 @@ Key assumptions include:
 !!! info "Note"
     The term **Redshift storage cost** refers to the cost of Redshift storage incurred for one month based on the corresponding RPS (Requests Per Second) specified in the above table. If the data is stored for more than one month, please refer to the [Redshift pricing](https://aws.amazon.com/redshift/pricing/?nc1=h_ls) for calculating the pricing.
 
+## Realtime streaming mode
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:122]: change to `Streaming mode`

File: docs/en/plan-deployment/cost.md [hunk 3]
```diff
@@ -119,6 +119,22 @@ Key assumptions include:
...
+
+If you enable real-time reports, the following additional cost will be incurred:
+Key assumptions include:
+
+- For 10/100 RPS, the batch EMR schedule interval is daily
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:127]: add a assumption: only one application

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by luorobin-a2z — APPROVED** [review]

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1560)
###### _updated: 7/31/2024, 07:00:18 (UTC+0)_

File: docs/en/plan-deployment/cost.md [hunk 4]
```diff
@@ -119,6 +119,31 @@ Key assumptions include:
...
+- For 1000 RPS, the batch EMR schedule interval is hourly
+
+| Request Per Second | Daily enable real-time duration |  Kinesis (On-Demand) | Redshift type     | Redshift cost  | Managed Apache Flink | Additional Total (USD/Month) |
+| ----------------------- | --------------------- | ---------------- | -------- | ------------------------ |  ----- | -----  |
+| 10RPS             | 1 Hour                |    $73    | Serverless (8 based RPU) |   $90       |  $6.8     |    $169.8  |
+|                         | 24 Hours                |    $207    | Serverless (8 based RPU) |    $1810      |  $163     |   $2180   |
+| 100RPS             | 1 Hour                |    $200    | Serverless (8 based RPU) |    $140      |  $10.2     |  $350.2    |
+|                         | 24 Hour                |    $1320    | Serverless (8 based RPU) |   $1950       |  $244.8     |  $3514.8    |
+| 1000RPS             | 1 Hour                |    $1189    | Serverless (16 based RPU) |   $75       |  $24     |   $1288   |
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:138]: Should be total Redshift cost(for batching and streaming)

File: docs/en/plan-deployment/cost.md [hunk 5]
```diff
@@ -119,6 +119,31 @@ Key assumptions include:
...
+
+If you enable real-time reports, the following additional cost will be incurred:
+
+Key assumptions include:
+
+- Only one application
+- For 10/100 RPS, the batch EMR schedule interval is daily
+- For 1000 RPS, the batch EMR schedule interval is hourly
+
+| Request Per Second | Daily enable real-time duration |  Kinesis (On-Demand) | Redshift type     | Redshift cost  | Managed Apache Flink | Additional Total (USD/Month) |
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:132]: Additional Total -> Total

File: docs/en/plan-deployment/cost.md [hunk 6]
```diff
@@ -119,6 +119,31 @@ Key assumptions include:
...
+|                         | 24 Hours                |    $207    | Serverless (8 based RPU) |    $1810      |  $163     |   $2180   |
+| 100RPS             | 1 Hour                |    $200    | Serverless (8 based RPU) |    $140      |  $10.2     |  $350.2    |
+|                         | 24 Hour                |    $1320    | Serverless (8 based RPU) |   $1950       |  $244.8     |  $3514.8    |
+| 1000RPS             | 1 Hour                |    $1189    | Serverless (16 based RPU) |   $75       |  $24     |   $1288   |
+
+!!! info "Tips"
+
+    - Kinesis (On-Demand) cost include ingestion Kinesis cost.
+
+    - If streaming is enabled 24 hours a day, provisioning redshift is more economical.
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:144]: Provisioned Redshift

File: docs/en/plan-deployment/cost.md [hunk 7]
```diff
@@ -119,6 +119,31 @@ Key assumptions include:
...
+| ----------------------- | --------------------- | ---------------- | -------- | ------------------------ |  ----- | -----  |
+| 10RPS             | 1 Hour                |    $73    | Serverless (8 based RPU) |   $90       |  $6.8     |    $169.8  |
+|                         | 24 Hours                |    $207    | Serverless (8 based RPU) |    $1810      |  $163     |   $2180   |
+| 100RPS             | 1 Hour                |    $200    | Serverless (8 based RPU) |    $140      |  $10.2     |  $350.2    |
+|                         | 24 Hour                |    $1320    | Serverless (8 based RPU) |   $1950       |  $244.8     |  $3514.8    |
+| 1000RPS             | 1 Hour                |    $1189    | Serverless (16 based RPU) |   $75       |  $24     |   $1288   |
+
+!!! info "Tips"
+
+    - Kinesis (On-Demand) cost include ingestion Kinesis cost.
```
Comment (zxkane) [inline docs/en/plan-deployment/cost.md:142]: -> Kinesis (On-Demand) costs include KDS used by the Ingestion module

**Review by zxkane — CHANGES_REQUESTED** [review]

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1560)
###### _updated: 8/1/2024, 03:45:35 (UTC+0)_

**Review by zxkane — APPROVED** [review]

**Comment by aws-solution-clickstream-analytics** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @dengmingtong on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.50|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|10|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1560)
###### _updated: 8/1/2024, 06:10:46 (UTC+0)_

**Pull request merged**